jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/JohnsonShortestPaths.java

```diff
@@ -161,12 +197,31 @@ private void run()
     private void runWithPositiveEdgeWeights(Graph<V, E> g)
     {
         /*
-         * Run Dijkstra for all vertices.
+         * Create vertex numbering for array representation of results.
+         */
+        vertexIndices = computeVertexIndices(g);
+        final int n = g.vertexSet().size();
+        distance = new double[n][n];
+        pred = new Object[n][n];
```

Comment:
Rather than doing typecasting at each point of access, wouldn't it be better to just do it once at array level here?